recirq/time_crystals/dtctask.py

```diff
@@ -0,0 +1,254 @@
+# Copyright 2021 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cirq
+import recirq
+import datetime
+import itertools
+import numpy as np
+from typing import Sequence, Optional, Dict
+import os
+
+EXPERIMENT_NAME = "time_crystals"
+DEFAULT_BASE_DIR = os.path.expanduser(f'~/cirq_results/{EXPERIMENT_NAME}')
+
+
+@recirq.json_serializable_dataclass(namespace='recirq.readout_scan',
+                                    registry=recirq.Registry,
+                                    frozen=False)
+class CompareDTCTask:
+    """ A task for managing inputs to a comparison Discrete Time Crystal experiment, comparing different options for parameters
+
+    Attributes
+        dataset_id: unique identifier for this dataset
+        qubits: chain of connected qubits available for the circuit
+        cycles: number of DTC cycles to consider for circuits
+        circuit_list: symbolic DTC circuit list
+        disorder_instances: number of disorder instances averaged over
+        options_dict: dict mapping DTCTask attribute names to options for that attribute, to take a product over
```

Comment:
cartesian product?

Reply:
what is the value of these dictionaries. the type annotation says its a numpy array. how is this an "option"?